src/NServiceBus.Hosting.Windows/StartableAndStoppable/IWantToRunWhenEndpointStartsAndStops.cs

```diff
@@ -0,0 +1,27 @@
+﻿namespace NServiceBus
+{
+    using System.Threading.Tasks;
+    using JetBrains.Annotations;
+
+    /// <summary>
+    /// Implementers will be invoked when the endpoint starts up.
+    /// Dependency injection is provided for these types.
+    /// </summary>
+    [UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
+    public interface IWantToRunWhenEndpointStartsAndStops
+    {
+        /// <summary>
+        /// Method called at startup.
```

Comment:
Should we spell out that users should not do blocking operations here? (since in v5 they could without any bad effects - http://docs.particular.net/nservicebus/lifecycle/iwanttorunwhenbusstartsandstops#versions-5-and-below )